airbyte-webapp/src/views/Connection/ConnectionForm/formConfigHelpers.ts

```diff
@@ -0,0 +1,88 @@
+import isEmpty from "lodash/isEmpty";
+
+import {
+  AirbyteStreamConfiguration,
+  DestinationSyncMode,
+  SyncMode,
+  SyncSchemaStream,
+} from "../../../core/domain/catalog";
+
+const getDefaultCursorField = (streamNode: SyncSchemaStream): string[] => {
+  if (streamNode.stream.defaultCursorField.length) {
+    return streamNode.stream.defaultCursorField;
+  }
+  return streamNode.config.cursorField;
+};
+
+const verifySupportedSyncModes = (streamNode: SyncSchemaStream): SyncSchemaStream => {
+  const {
+    stream: { supportedSyncModes },
+  } = streamNode;
+
+  if (!isEmpty(supportedSyncModes)) return streamNode;
+  return { ...streamNode, stream: { ...streamNode.stream, supportedSyncModes: [SyncMode.FullRefresh] } };
+};
+
+const verifyConfigCursorField = (streamNode: SyncSchemaStream): SyncSchemaStream => {
+  const { config } = streamNode;
+
+  return {
+    ...streamNode,
+    config: {
+      ...config,
+      cursorField: !isEmpty(config.cursorField) ? config.cursorField : getDefaultCursorField(streamNode),
```

Comment:
```suggestion
cursorField: config.cursorField.length ? config.cursorField : getDefaultCursorField(streamNode),
```
Same here, we know this is a `string[]`

Reply:
Ok, will fix it as well